File: email-backup.java/src/main/java/demo/Flags.java
package demo;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

final public class Flags {

    final private String email;
    final private String password;
    final private String host;
    final private int port;
    final private List<String> excludedExtensions;
    final private List<String> includedFiles;
    final private List<String> emailsTo;

    public Flags(String email, String password, String host, int port, String[] excludedExtensions, String[] includedFiles, String[] emailsTo) {
        this.email = email;
        this.password = password;
        this.host = host;
        this.port = port;
        this.excludedExtensions = toList(excludedExtensions);
        this.includedFiles = toList(includedFiles);
        this.emailsTo = toList(emailsTo);
    }

    private List<String> toList(String[] array) {
        List<String> list = new ArrayList<>();
        if (array != null) {
            list.addAll(Arrays.asList(array));
        }
        return list;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public String getHost() {
        return host;
    }

    public int getPort() {
        return port;
    }

    public List<String> getExcludedExtensions() {
        return excludedExtensions;
    }

    public List<String> getIncludedFiles() {
        return includedFiles;
    }

    public List<String> getEmailsTo() {
        return emailsTo;
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/Flags.kt
package demo

public class Flags(public val email: String, public val password: String, public val host: String, public val port: Int, excludedExtensions: Array<String>?, includedFiles: Array<String>?, emailsTo: Array<String>?) {

    public val excludedExtensions: List<String>
    public val includedFiles: List<String>
    public val emailsTo: List<String>

    init {
        this.excludedExtensions = toList(excludedExtensions)
        this.includedFiles = toList(includedFiles)
        this.emailsTo = toList(emailsTo)
    }

    private fun toList(array: Array<String>?): List<String> {
        return if (array == null) listOf() else listOf(*array)
    }
}
